src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1640,13 +1639,11 @@ private void RewriteArgumentsForComCall(
 
                 actualArguments[argIndex] = new BoundSequence(
                     argument.Syntax,
-                    locals: ImmutableArray<LocalSymbol>.Empty,
-                    sideEffects: ImmutableArray.Create<BoundExpression>(boundAssignmentToTemp),
+                    locals: [boundTemp.LocalSymbol],
```

Comment:
> It looks like we are going to pass a reference to a local that goes out of scope.

True, but even without the change, the temp might live too short - see test `RefTemp_Escapes_ComCall`. Moving the temp into the sequence inside the call ensures the emit layer does not free the temp after it detects the call might escape temps. If the temp is declared outside the call, the emit layer does not see that it is a part of a "might escape" call and that its freeing should be avoided.

It turns out this change is what causes the locals being declared twice in one lowered tree. But I imagine this can be fixed in the callers of this method.

> But that is likely to lead to a worse IL

The mentioned test verifies IL and at least in that scenario it does not look worse.